server/api/resolvers/donorHomepageResolvers.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    Banner,
+    DonorHomepage,
+    DonorHomepageInterface,
+    Map,
+    Statistic,
+    StatisticMeasurement,
+    Testimonial,
+    TestimonialCarousel
+} from "../../database/models/donorHomepageModel";
+
+const getDonorHomepageData = async <Type>(property: string): Promise<Type> => {
+    return DonorHomepage.find()
+        .exec()
+        .then((donorHomepageData) => {
+            return donorHomepageData[0][property];
+        });
+};
+
+const donorHomepageQueryResolvers = {
+    donorHomepageBanner: async (): Promise<Banner> => {
+        return getDonorHomepageData<Banner>("banner");
+    },
+    donorHomepageTestimonialCarousel: async (): Promise<TestimonialCarousel> => {
+        return getDonorHomepageData<TestimonialCarousel>("testimonialCarousel");
+    },
+    donorHomepageMap: async (): Promise<Map> => {
+        return getDonorHomepageData<Map>("map");
+    },
+    donorHomepageStatistics: async (): Promise<Array<Statistic>> => {
+        return getDonorHomepageData<Array<Statistic>>("statistics");
+    }
+};
+
+const donorHomepageMutationResolvers = {
+    updateDonorHomepage: async (
+        _,
+        { mapTestimonials, carouselTestimonials, statMeasurements },
+        { authenticateUser }
+    ): Promise<DonorHomepageInterface> => {
+        return authenticateUser().then(async () => {
+            const map = await donorHomepageQueryResolvers.donorHomepageMap();
+            map.testimonials = mapTestimonials;
+
+            const statistics = await donorHomepageQueryResolvers.donorHomepageStatistics();
+            statistics.forEach((statistic) => {
+                const type = statistic.type.toString();
+                if (statMeasurements[type] !== "") {
```

Comment:
i think we should also check if `statMeasurements[type]` is null, since any of the three stats could be missing from the input (e.g. if `statMeasurements = { REGULAR_DONOR: "250+" }`, then the other two are missing - in this case the statistic for the other two would be updated with a null value)